File: Try_GameEngine5/src/com/example/try_gameengine/viewport/MyMap.java
package com.example.try_gameengine.viewport;

import android.graphics.Point;
import android.graphics.PointF;
import android.widget.ImageView;

import com.example.try_gameengine.framework.CommonUtil;

public class MyMap {
	
	private static float totalX = 1;
	private static float totalY = 1;
	private static float displayX = 1;
	private static float displayY = 1;
	
	public static void setTotalXY(float totalX, float totalY){
		MyMap.totalX = totalX;
		MyMap.totalY = totalY;
	}
	
	public static void setDisplayXY(float displayX, float displayY){
		MyMap.displayX = displayX;
		MyMap.displayY = displayY;
	}
	
	public static PointF setImageXYByOldXY(int oldX, int oldY) {
		float newX = (float) (oldX / totalX) * displayX;
		float newY = (float) (oldY / totalY) * displayY;

		return new PointF(newX, newY);
	}
	
	public static Point getWH(int oldW, int oldH){
		Point point = new Point((int) ((oldW / totalX) * displayX), (int) ((oldH / totalY) * displayY));
		return point;
	}
	
	public static void setInfo(int orangeW, int orangeH, int displayW, int displayH){
		totalX = orangeW;
		totalY = orangeH;
		displayX = displayW;
		displayY = displayH;
	} 
}

File: Try_GameEngine5/src/com/example/try_gameengine/viewport/MyMap.kt
package com.example.try_gameengine.viewport

import android.graphics.Point
import android.graphics.PointF

object MyMap {
    private var totalX = 1f
    private var totalY = 1f
    private var displayX = 1f
    private var displayY = 1f

    fun setTotalXY(totalX: Float, totalY: Float) {
        MyMap.totalX = totalX
        MyMap.totalY = totalY
    }

    fun setDisplayXY(displayX: Float, displayY: Float) {
        MyMap.displayX = displayX
        MyMap.displayY = displayY
    }

    fun setImageXYByOldXY(oldX: Int, oldY: Int): PointF {
        val newX = (oldX / totalX) * displayX
        val newY = (oldY / totalY) * displayY

        return PointF(newX, newY)
    }

    fun getWH(oldW: Int, oldH: Int): Point {
        val point =
            Point(((oldW / totalX) * displayX).toInt(), ((oldH / totalY) * displayY).toInt())
        return point
    }

    fun setInfo(orangeW: Int, orangeH: Int, displayW: Int, displayH: Int) {
        totalX = orangeW.toFloat()
        totalY = orangeH.toFloat()
        displayX = displayW.toFloat()
        displayY = displayH.toFloat()
    }
}
